substrate/utils/frame/benchmarking-cli/src/pallet/writer.rs

```diff
@@ -175,6 +179,121 @@ fn map_results(
+		`DispatchClass::Normal` extrinsics) block. \n\nProvided values for the maximum extrinsic weight: \
+		{:?} (ref_time), {:?} (proof_size)\n\n<u>Results:</>\n",
+		max_extrinsic_weight.ref_time(),
+		max_extrinsic_weight.proof_size()
+	);
+	let mut sanity_weight_check_passed = true;
+	// Loop through all benchmark results.
+	for ((pallet, instance), results) in all_results.iter() {
+		println!("Pallet: {}\nInstance: {}\n", pallet, instance);
+		for result in results {
+			// Constant `ref_time` & `pov_size`.
+			let mut total_weight = Weight::from_parts(
+				result.base_weight.try_into().unwrap(),
+				result.base_calculated_proof_size.try_into().unwrap(),
+			);
+			// `ref_time` multiplied by complexity parameter.
+			for component in &result.component_weight {
+				total_weight.saturating_accrue(
+					Weight::from_parts(component.slope.try_into().unwrap(), 0)
+						.saturating_mul(max_component(&component, &result.component_ranges)),
+				);
+			}
+			// Constant storage reads.
+			total_weight.saturating_accrue(db_weight.reads(result.base_reads.try_into().unwrap()));
+
+			// Storage reads multiplied by complexity parameter.
+			for component in &result.component_reads {
+				total_weight.saturating_accrue(
+					db_weight
+						.reads(component.slope.try_into().unwrap())
+						.saturating_mul(max_component(&component, &result.component_ranges)),
+				);
+			}
+			// Constant storage writes.
+			total_weight
+				.saturating_accrue(db_weight.writes(result.base_writes.try_into().unwrap()));
+
+			// Storage writes multiplied by complexity parameter.
+			for component in &result.component_writes {
+				total_weight.saturating_accrue(
+					db_weight
+						.writes(component.slope.try_into().unwrap())
+						.saturating_mul(max_component(&component, &result.component_ranges)),
+				);
+			}
+			// `pov_size` multiplied by complexity parameter.
+			for component in &result.component_calculated_proof_size {
+				total_weight.saturating_accrue(
+					Weight::from_parts(0, component.slope.try_into().unwrap())
+						.saturating_mul(max_component(&component, &result.component_ranges)),
+				);
+			}
+			// Comparing (worst case scenario) the extrinsic weight against the maximum extrinsic
+			// weight.
+			if total_weight.ref_time() > max_extrinsic_weight.ref_time() ||
+				total_weight.proof_size() > max_extrinsic_weight.proof_size()
+			{
+				sanity_weight_check_passed = false;
+				color_print::cprintln!("<s,r>WARNING!!!</>",);
+				color_print::cprintln!(
+					"<r>The following extrinsic exceeds the maximum extrinsic weight:</>",
+				);
+			}
+			color_print::cprintln!("- <s>'{}'</>: {:?}\nPercentage of max. extrinsic weight: {:.2}% (ref_time), {:.2}% (proof_size)\n", 
+				result.name,
+				total_weight,
+				(total_weight.ref_time() as f64 / max_extrinsic_weight.ref_time() as f64) * 100.0,
+				(total_weight.proof_size() as f64 / max_extrinsic_weight.proof_size() as f64) * 100.0,
+			);
```

Comment:
Just one `log::warn!` is enough per extrinsic. Looks like the previous code is a bit overkill.